pytorch_lightning/plugins/environments/lightning_environment.py

```diff
@@ -40,13 +40,15 @@ def __init__(self):
         self._global_rank: int = 0
         self._world_size: int = 1
 
-    def creates_children(self) -> bool:
-        """Returns whether the cluster creates the processes or not.
+        # If at least `LOCAL_RANK` is available as environment variable, Lightning assumes the user acts as the
+        # process launcher/job scheduler and Lightning will not launch new processes.
+        # We capture this on initialization such that the value of `creates_children` doesn't fluctuate
+        # during the course of execution
+        self._creates_children: bool = "LOCAL_RANK" in os.environ
 
-        If at least :code:`LOCAL_RANK` is available as environment variable, Lightning assumes the user acts as the
-        process launcher/job scheduler and Lightning will not launch new processes.
-        """
-        return "LOCAL_RANK" in os.environ
+    def creates_children(self) -> bool:
+        """Returns whether the cluster creates the processes or not."""
+        return self._creates_children
```

Comment:
note to reviewer: this will return different values on different ranks, which makes process reconciliation with this environment harder. on local rank 0, this will be false, meaning we should run process reconciliation. on other local rank > 0, this will be True, meaning we shouldn't run process reconciliation. 

we could add logic like this:
```
(if self.local_rank == 0 and not cluster_environment.creates_children()) or if self.local_rank > 0 and cluster_environment.creates_children()):
    # proceed with reconciliation
```

but this is going to break for torchelastic/SLURM where cluster_environment.creates_children() is always True, as we'll have different ranks participating in the reconciliation, which will lead to its own hangs. 

the only solution there is to special-case the lightning environment, but that's already happening with `call_children_scripts` above, so it doesn't make sense to repeat that logic here